common/src/main/java/org/openmrs/analytics/GcpStoreUtil.java

```diff
@@ -42,21 +42,16 @@
  *
  * @see org.openmrs.analytics.FhirStoreUtil
  */
-public class GcpStoreUtil extends FhirStoreUtil {
+class GcpStoreUtil extends FhirStoreUtil {
 	
 	private static final Logger log = LoggerFactory.getLogger(GcpStoreUtil.class);
 	
 	private static final GsonFactory JSON_FACTORY = new GsonFactory();
 	
 	private static final NetHttpTransport HTTP_TRANSPORT = new NetHttpTransport();
 	
-	GcpStoreUtil(String gcpFhirStore, IRestfulClientFactory clientFactory) throws IllegalArgumentException {
-		super(gcpFhirStore, clientFactory);
-		
-		if (!matchesGcpPattern(gcpFhirStore)) {
-			throw new IllegalArgumentException(
-			        String.format("The gcpFhirStore %s does not match Google cloud fhir dataset pattern!", gcpFhirStore));
-		}
+	protected GcpStoreUtil(String sinkUrl, String sinkUsername, String sinkPassword, IRestfulClientFactory clientFactory) {
```

Comment:
I don't think BasicAuth makes sense for the GCP FHIR store and regardless we are not using it here. So I suggest that you drop username/password from the constructor signature here to make it more clear and prevent confusion.